Modules/StudyProgramme/classes/model/Progress/class.ilStudyProgrammeProgress.php

```diff
@@ -337,153 +375,250 @@ public function getLastChangeBy()
     
     /**
      * Set the id of the user who did the last change on this progress.
-     *
-     * Throws when $a_usr_id is not the id of a user.
-     *
-     * @throws ilException
-     * @return $this
      */
-    public function setLastChangeBy(int $a_usr_id = null) : ilStudyProgrammeProgress
+    public function withLastChangeBy(int $usr_id) : ilStudyProgrammeProgress
```

Comment:
Ich denke hier gilt das gleiche wie bei den nämlichen Methoden in der Assignment-Klasse.